src/Sulu/Bundle/AdminBundle/Controller/BlockIdController.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\AdminBundle\Controller;
+
+use Sulu\Bundle\AdminBundle\Application\BlockIdGenerator\BlockIdGeneratorInterface;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Controller for generating unique block IDs.
+ * Used by the frontend to generate IDs when creating or duplicating blocks.
+ */
+class BlockIdController
+{
+    public function __construct(
+        private readonly BlockIdGeneratorInterface $blockIdGenerator,
+    ) {
+    }
+
+    public function generateAction(): Response
+    {
+        $id = $this->blockIdGenerator->generateId();
+
+        return new JsonResponse(['id' => $id]);
```

Comment:
In case of copy blocks we do now a request per block? Should we keep in mind todo a single request here if possible with `?length=...`

```suggestion
        $blockIds = [];
        $request->query->getInt('length', 1)
        
        for ($i = 0; $i<=$length;++$i) {
            $blockIds = ['id' => $this->blockIdGenerator->generateId()]; // maybe already return it in this format make it easier to merge it together locally
        }

        return new JsonResponse(['_embedded' => ['blockIds' => $blockIds]); // not sure about this in a post Action format
```

If you think we should keep it as a single id endpoint I'm fine with it.